napari/components/viewer_model.py

```diff
@@ -492,16 +496,17 @@ def _on_add_layer(self, event):
         self._layer_help_from_mode(layer)
 
         # Update dims and grid model
-        self._on_layers_change()
+        self._on_layers_change(add_layer=True)
```

Comment:
It seems like we added the `add_layer` and `restore_point` parameters just to handle the case when we're going from 0 to 1 layers. In that case, we don't restore the current point because we know it's really just a dummy point when there are no layers and want to skip slicing cause by effectively changing `current_point`. But then we call `_update_layers` and slice anyway. And then we'll slice again with the midpoint in the code below.

I'd rather remove both of those parameters (to increase readability) and just slice 3 times (knowing that the second time should be fast anyway because nothing changed). What do you think?

Reply:
I agree and it happens very rarely, I added it to satisfy the `napari/layers/_tests/test_dask_layers.py` tests. It seems to check how often the set_point method is called on load. So I adjusted this tests :)